enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,323 @@
+
+### Version Skew Strategy
+N/A
+
+### Operational Aspects of API Extensions
+N/A - no api extensions are being introduced in this proposal
+
+#### Failure Modes
+N/A - no api extensions are being introduced in this proposal
+
+#### Support Procedures
+N/A - no api extensions are being introduced in this proposal
+
+## Implementation History
+N/A
+
+## Alternatives
+
+### Implement the ingress capability for the standalone OpenShift
+
+The goal of this enhancement is to satisfy the HyperShift's requirements. The implementation of the ingress capability for the standalone OpenShift brings a lot of additional complexity, unnecessary for HyperShift. Most probably the following components would need to be updated to tolerate the absence of the ingress operator:
+- The openshift installer. It [checks for the admission of the console route](https://github.com/wking/openshift-installer/blob/3d49ffbff57011bbf846bf91de11526bb3543a83/cmd/openshift-install/create.go#L543) by the ingress controller.
+- The authentication operator. It checks the availability of [the authentication server's](https://github.com/openshift/cluster-authentication-operator/blob/da39951a53ad95be28e32a48f278cc23f41e99a7/pkg/controllers/oauthendpoints/oauth_endpoints_controller.go#L30-L31) 
+and [console](https://github.com/openshift/cluster-authentication-operator/blob/da39951a53ad95be28e32a48f278cc23f41e99a7/pkg/controllers/configobservation/console/observe_consoleurl.go#L15) routes.
+- [The component routes API](https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/ingress-config-openshift-io-v1.html#status-componentroutes). The ingress operator is [responsible for the RBAC](https://github.com/openshift/cluster-ingress-operator/blob/8f98c618c5609cf7fcb97e5c61dc1e7a30576925/pkg/operator/controller/configurable-route/controller.go#L42-L52)
+which the component operators need to be able to access the custom TLS certificates from `openshift-config` namespace.
+
+**Note**: it's difficult to anticipate the impact of the ingress operator removal on the installation process without the actual capability implementation.
+
+### Allow standalone OpenShift to be broken if the ingress is disabled
+
+This approach implies the implementation of the ingress capability the way it's described in [how to implement a new capability](/enhancements/installer/component-selection.md#how-to-implement-a-new-capability).
+But unlike this enhancement, it doesn't have a goal to tolerate the absence of the ingress operator.
+That is, anything which can break during or after the cluster installation is allowed to break.
+This approach has a clear downside of the postponed failure which questions the correctness of the capability implementation from the user's point of view.
```

Comment:
In other words, we should avoid giving the user a knob that breaks the cluster, right?

Reply:
Yes. Rephrased this one to:

> This approach has a notable downside: it provides the user with a means to break the cluster, raising concerns about the correctness of the capability implementation from the user's perspective.